index.js

```diff
@@ -1,211 +1,73 @@
 'use strict';
-var path = require('path');
-var url = require('url');
-var caw = require('caw');
-var concatStream = require('concat-stream');
-var decompress = require('gulp-decompress');
-var eachAsync = require('each-async');
-var filenamify = require('filenamify');
-var got = require('got');
-var isUrl = require('is-url');
-var objectAssign = require('object-assign');
-var readAllStream = require('read-all-stream');
-var rename = require('gulp-rename');
-var streamCombiner = require('stream-combiner2');
-var PassThrough = require('readable-stream/passthrough');
-var Vinyl = require('vinyl');
-var vinylFs = require('vinyl-fs');
-var Ware = require('ware');
-
-/**
- * Initialize a new `Download`
- *
- * @param {Object} opts
- * @api public
- */
-
-function Download(opts) {
-	if (!(this instanceof Download)) {
-		return new Download(opts);
-	}
-
-	this.opts = objectAssign({encoding: null}, opts);
-	this.ware = new Ware();
-}
-
-module.exports = Download;
-
-/**
- * Get or set URL to download
- *
- * @param {String} url
- * @param {String} dest
- * @api public
- */
-
-Download.prototype.get = function (url, dest) {
-	if (!arguments.length) {
-		return this._get;
-	}
-
-	this._get = this._get || [];
-	this._get.push({
-		url: url,
-		dest: dest
+const fs = require('fs');
+const path = require('path');
+const url = require('url');
+const caw = require('caw');
+const Decompress = require('decompress');
+const filenamify = require('filenamify');
+const getStream = require('get-stream');
+const got = require('got');
+const mkdirp = require('mkdirp');
+const pify = require('pify');
+const fsP = pify(fs);
+
+const createPromise = (uri, output, stream, opts) => new Promise((resolve, reject) => {
+	stream.on('response', res => {
```

Comment:
After looking into it again, I don't think there is that much benefit. You still need the wrapper promise.

```js
rfpify(stream.on.bind(stream)('response')
    .then(res => opts.encoding === null ? getStream.buffer(res) : getStream(res, opts))
    .then(resolve)
    .catch(reject);

stream.on('error', reject);
```